### PR TITLE
Lji 16 employee filters

### PR DESCRIPTION
**Pull request by Cioperis**
1. added employee filtering by first name, last name, user name, ssn, date of birth and status
2. default sorting by date of birth
3. added a validation check tool (if entity is null, empty)
4. new endpoint /filtered
5. new postman collection
6. added /filtered tests
7. deleted duplicate employee tables

### CODE REVIEW

**Comment by SavaLX** [comment]
The branch's name should start from feature/, bugfix/, hotfix/, etc.

File: src/main/java/com/coherentsolutions/pot/insurance/controller/EmployeeController.java [hunk 1]
```diff
@@ -23,6 +27,14 @@
 public class EmployeeController {
     private final EmployeeService employeeService;
 
+    @GetMapping("/filtered")
+    @ResponseStatus(HttpStatus.OK)
+    public Page<EmployeeDTO> filterAndSortEmployees(
```
Comment (SavaLX) [inline src/main/java/com/coherentsolutions/pot/insurance/controller/EmployeeController.java:32]: Let's use the same approach in the naming of controllers method
```suggestion
    public Page<EmployeeDTO> getFilteredSortedEmployees(
```

File: src/main/java/com/coherentsolutions/pot/insurance/service/EmployeeService.java [hunk 2]
```diff
@@ -16,6 +29,42 @@
 public class EmployeeService {
     private final EmployeeRepository employeeRepository;
 
+    private Specification<EmployeeEntity> buildSpecification(EmployeeFilterCriteria employeeFilterCriteria) {
```
Comment (SavaLX) [inline src/main/java/com/coherentsolutions/pot/insurance/service/EmployeeService.java:32]: Move private methods to the bottom

File: src/test/java/com/coherentsolutions/pot/insurance/employee/EmployeeControllerTest.java [hunk 3]
```diff
@@ -134,4 +140,35 @@ void testGetEmployee_NotFound() throws Exception {
 
     verify(employeeService).getEmployee(id);
   }
+
+  @Test
+  @WithMockUser(username = "admin")
+  void testFilterAndSortEmployees() throws Exception {
```
Comment (SavaLX) [inline src/test/java/com/coherentsolutions/pot/insurance/employee/EmployeeControllerTest.java:146]: Please reformat fully your test class regarding this comment - https://github.com/Java-POT-Summer-24/pot-java-summer-24/pull/24#discussion_r1665361282

**Comment by SavaLX** [comment]
Cover new features by unit-tests

**Review by SavaLX — CHANGES_REQUESTED** [review]

File: src/main/java/com/coherentsolutions/pot/insurance/controller/EmployeeController.java (hunk 1 above)
Comment (Cioperis) [reply src/main/java/com/coherentsolutions/pot/insurance/controller/EmployeeController.java]: updated namings

File: src/test/java/com/coherentsolutions/pot/insurance/employee/EmployeeControllerTest.java (hunk 3 above)
Comment (Cioperis) [reply src/test/java/com/coherentsolutions/pot/insurance/employee/EmployeeControllerTest.java:146]: fully updated tests class

Comment (SavaLX) [reply src/test/java/com/coherentsolutions/pot/insurance/employee/EmployeeControllerTest.java:146]: Please take a look at this PR - https://github.com/Java-POT-Summer-24/pot-java-summer-24/pull/26/files#diff-5f4041addff00595da8b10f7e613a5f6c0bf7a553c06f632be1679f1e8a9f636 .
There are examples of unit and integration tests

**Comment by SavaLX** [comment]
Fix build

File: src/main/resources/changelog/dml/6_insert_employees_table.xml [hunk 4]
```diff
@@ -9,7 +9,7 @@
     <sql>
       INSERT INTO employee (id, first_name, last_name, user_name, email, date_of_birth, ssn, phone_number, status)
       VALUES
-      (RANDOM_UUID(), 'John', 'Doe', 'johndoe', 'john.doe@example.com', '1980-01-01', '123456789', '1234567890', 'ACTIVE'),
+      ('d5e38eb0-58a3-4a0f-878c-72eac1877d5e', 'John', 'Doe', 'johndoe', 'john.doe@example.com', '1980-01-01', '123456789', '1234567890', 'ACTIVE'),
```
Comment (NormantasN) [inline src/main/resources/changelog/dml/6_insert_employees_table.xml:12]: Refer to [#26 (comment)](https://github.com/Java-POT-Summer-24/pot-java-summer-24/pull/26#discussion_r1665848645)

Comment (Cioperis) [reply src/main/resources/changelog/dml/6_insert_employees_table.xml]: Done

**Pull request closed without merging**